kernel/thread/worker.c

```diff
@@ -89,13 +90,20 @@ void thd_worker_wakeup(kthread_worker_t *worker) {
 }
 
 void thd_worker_destroy(kthread_worker_t *worker) {
+    uint32_t flags;
+    kthread_t *thd;
+
     assert(worker != NULL);
 
+    flags = irq_disable();
+
     worker->quit = true;
     genwait_wake_one(worker);
+    thd = worker->thd;
```

Comment:
We are caching this to avoid the chance that once irqs are reenabled the worker thread finishes all on its own and worker is now freed before calling `thd_join`